### DISCUSSION TITLE
Accessing Copilot Endpoint Allowed?

### DISCUSSION THREAD

**Discussion by zbirenbaum** [community/community#39278] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

~~Are there any plans to release an API for accessing the copilot endpoint, or open sourcing the copilot agent so that people can extend and make improvements to it?~~ If not, if I were to implement a way of authenticating and accessing the endpoint outside of the distributed copilot extension source would it get shut it down?

~~I'm the developer of a few fairly popular neovim plugins for github copilot, and the current radio silence on this issue has frustrated a number of people on this discussion board, many of my users, and to be perfectly honest myself as well. It seems like having access be more open would increase the number of use cases dramatically, which would also boost subscriptions a ton, while leaving Microsoft/Github perfectly in control of the proprietary model....~~

~~Very often my users request features for things like what is available in copilot labs, which aren't available as part of vanilla copilot. I have also seen numerous requests in these discussions for a way to access copilot (as a service they are paying for) in order to implement things known to be possible with the Codex models it is based on like cli completions. I wrote a very basic LSP for Codex I planned to use to support those types of features, but ran into rate limiting issues that make using it like copilot infeasible.~~

**@davecheney Thank you for the response but can you provide any comment on whether or not creating my own api for accessing the endpoint would get shut down?**

**Comment by charliie-dev** [comment]
upvoted

**Comment by davecheney** [comment]
> Are there any plans to release an API for accessing the copilot endpoint

We have no plans to offer a public API endpoint at this time.
